imagepipeline-base/src/main/java/com/facebook/cache/disk/DiskStorageCache.java

```diff
@@ -592,10 +592,16 @@ private void updateFileCacheSizeLimit() {
   }
 
   public long getSize() {
+    if (!mCacheStats.isInitialized()){
+      maybeUpdateFileCacheSize();
```

Comment:
We would have to have a `synchronized(mLock)` around this, as `maybeUpdateFileCacheSize()` is expected to be called in a synchronized manner (see line 703/709).

Alternatively, we could throw an `IllegalStateException` in those cases where the client has to wait for the initialization.

Which variant would you prefer? And what the use-cases would you anticipate where clients need to check this early on?